sympy/core/add.py

```diff
@@ -70,6 +70,83 @@ def _unevaluated_Add(*args):
 
 
 class Add(Expr, AssocOp):
+    """
+    Expression representing addition operation for algebraic group.
+    Infix operator ``+`` on most ``Expr`` objects, except the one between
+    numbers, call this class.
+
+    This class mainly deals with the addition over the field of complex
+    numbers. ``Add(*args)`` is more efficient than ``sum(args)`` because
+    the algoritm is linear in the number of terms.
```

Comment:
These two sentences don't really belong together. Perhaps just remove the first.

It's probably worth explicitly clarifying that
```
expr1 + expr2    is equivalent to Add(expr1, expr2)
```
and therefore that
```
expr1 + expr2 + expr3  is Add(expr1, Add(expr2, expr3))
```
and since `sum` is implemented using `+` then
```
sum(args) is Add(args[0], Add(args[1], Add(args[2], ...))
```
Since each call to `Add` has a linear complexity in the number of terms `sum(args)` has quadratic complexity whereas `Add(*args)` is linear.